packages/utils/src/tests/test_fuzz_checkpoint.cairo

```diff
@@ -0,0 +1,122 @@
+    let checkpoints = build_checkpoints_array(len, key_step);
+
+    let mut expected_prev = 0;
+    for point in checkpoints {
+        let (prev, new) = mock_trace.push_checkpoint(*point.key, *point.value);
+        assert_eq!(prev, expected_prev);
+        assert_eq!(new, *point.value);
+        expected_prev = new;
+    };
+    assert_eq!(mock_trace.get_length(), len);
+}
+
+#[test]
+fn test_upper_lookup(len_seed: u64, key_step_seed: u64) {
+    let len = 2 + len_seed % 99; // [2..100]
+    let key_step = 1 + key_step_seed % 1_000_000; // [1..1_000_000]
+
+    let mut mock_trace = CONTRACT_STATE();
+    let checkpoints = build_checkpoints_array(len, key_step);
+    push_checkpoints(checkpoints);
+
+    for i in 0..len {
+        let index = i.try_into().unwrap();
+        let checkpoint = *checkpoints.at(index);
+        let found_value = mock_trace.upper_lookup(checkpoint.key);
+        assert_eq!(found_value, checkpoint.value);
+    };
+}
+
+#[test]
+fn test_upper_lookup_recent(len_seed: u64, key_step_seed: u64) {
+    let len = 2 + len_seed % 99; // [2..100]
+    let key_step = 1 + key_step_seed % 1_000_000; // [1..1_000_000]
+
+    let mut mock_trace = CONTRACT_STATE();
+    let checkpoints = build_checkpoints_array(len, key_step);
+    push_checkpoints(checkpoints);
+
+    for i in 0..len {
+        let index = i.try_into().unwrap();
+        let checkpoint = *checkpoints.at(index);
+        let found_value = mock_trace.upper_lookup_recent(checkpoint.key);
+        assert_eq!(found_value, checkpoint.value);
+    };
+}
+
+#[test]
+fn test_get_at_position(len_seed: u64, key_step_seed: u64) {
+    let len = 2 + len_seed % 99; // [2..100]
+    let key_step = 1 + key_step_seed % 1_000_000; // [1..1_000_000]
+
+    let mut mock_trace = CONTRACT_STATE();
+    let checkpoints = build_checkpoints_array(len, key_step);
+    push_checkpoints(checkpoints);
+
+    for i in 0..len {
+        let index = i.try_into().unwrap();
+        let checkpoint = *checkpoints.at(index);
+        let found_checkpoint = mock_trace.get_at_position(i);
+        assert!(found_checkpoint == checkpoint);
+    };
+}
+
+#[test]
+#[should_panic(expected: 'Vec overflow')]
+fn test_at_position_out_of_bounds(len_seed: u64, key_step_seed: u64) {
+    let len = 2 + len_seed % 99; // [2..100]
+    let key_step = 1 + key_step_seed % 1_000_000; // [1..1_000_000]
+
+    let mut mock_trace = CONTRACT_STATE();
+    let checkpoints = build_checkpoints_array(len, key_step);
+    push_checkpoints(checkpoints);
+
+    mock_trace.get_at_position(len);
+}
+
+#[test]
+fn test_pack_unpack(key: u64, value: u256) {
+    let initial_checkpoint = Checkpoint { key, value };
+
+    let packed_value = StorePacking::pack(initial_checkpoint);
+    let unpacked_checkpoint: Checkpoint = StorePacking::unpack(packed_value);
+
+    assert!(initial_checkpoint == unpacked_checkpoint);
+}
+
+//
+// Helpers
+//
+
+fn build_checkpoints_array(len: u64, key_step: u64) -> Span<Checkpoint> {
```

Comment:
We now can implement the Fuzzable trait for checkpoint for tests to avoid these helpers improving clarity.